src/api/node/sse/client.rs

```diff
@@ -0,0 +1,133 @@
+use super::{
+    error::ClientError,
+    types::{CoreCommand, EventType},
+    ClientCore, SseData,
+};
+use std::time::Duration;
+use tokio::sync::{mpsc, oneshot};
+
+pub struct Client {
+    command_sender: mpsc::Sender<CoreCommand>,
+}
+
+impl Client {
+    pub async fn new(url: &str) -> Self {
+        let client_core = ClientCore::new(url).await;
+
+        let (tx, rx) = mpsc::channel(32);
+        //TODO: not sure how to handle these errors from run_client_core
+        let _handle = tokio::spawn(async move {
+            let _ = run_client_core(rx, client_core).await;
```

Comment:
In similar case I am logging every transient error, and panicking on unrecoverable/unexpected ones - [see here](https://github.com/cspr-rad/kairos/blob/4ac6aee4f64e4a5d6bca3f1b2770f5090dff9c2d/kairos-server/src/l1_sync/service.rs#L70-L73). In this case I would simply log error and panic from task.

Reply:
Done